common/src/main/java/tachyon/security/authorization/FileSystemPermission.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.authorization;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.ExceptionMessage;
+
+/**
+ * A class for file/directory permissions.
+ */
+public final class FileSystemPermission {
+  //POSIX permission style
+  private FileSystemAction mUseraction;
+  private FileSystemAction mGroupaction;
+  private FileSystemAction mOtheraction;
+
+  /**
+   * Constructs an instance of {@link FileSystemPermission} with the given {@link FileSystemAction}.
+   * @param userFileSystemAction user action
+   * @param groupFileSystemAction group action
+   * @param otherFileSystemAction other action
+   */
+  public FileSystemPermission(FileSystemAction userFileSystemAction, FileSystemAction
+      groupFileSystemAction, FileSystemAction otherFileSystemAction) {
+    set(userFileSystemAction, groupFileSystemAction, otherFileSystemAction);
+  }
+
+  /**
+   * Constructs an instance of {@link FileSystemPermission} with the given mode.
+   * @param mode
+   * @see #toShort()
+   */
+  public FileSystemPermission(short mode) {
+    fromShort(mode);
+  }
+
+  /**
+   * Copy constructor.
+   * @param otherPermission
+   */
+  public FileSystemPermission(FileSystemPermission otherPermission) {
+    set(otherPermission.mUseraction, otherPermission.mGroupaction, otherPermission.mOtheraction);
+  }
+
+  /**
+   * Get user action
```

Comment:
either do:

```
Gets user action.

@return ...
```

or simply

```
@return ...
```

as the former does not add any new information